torch/types.py

```diff
@@ -70,8 +70,10 @@
 
 # Meta-type for "device-like" things.  Not to be confused with 'device' (a
 # literal device object).  This nomenclature is consistent with PythonArgParser.
+ExplicitDevice: TypeAlias = Union[_device, str, int]
```

Comment:
What is the benefit of having this "ExplicitDevice" and "Device" if they're the same thing?

Reply:
Device can be None, which is cpu. But for accelerator APIs, their devices are accelerator cards (CUDA,XPU,...).
ExplicitDevice means that users can't pass None (CPU) to these APIs, that is meaningless..

Reply:
> Device can be None, which is cpu.

I don't understand what you mean by that? Where did you read that?
Generally, when passing an optional device around, empty means the current default device not None.
In current_accelerator() API, we use None when no accelerator is available. But that's very unique to only that one function.

Reply:
Sorry, misunderstanding..

Reply:
ExplicitDevice is used in only one place `set_device_index`, here the device shouldn't be None.